Automattic/wp-calypso
# Add skip handling to email signup step

#### Changes proposed in this Pull Request

* This PR wires in two important features to the newly created email step:
1.  When a user picks a free WordPress.com subdomain from the domains step, we don't show the new email step
2. We correctly handle the user clicking on the "Skip" button in the new email step

This change doesn't take on the work to add Professional Email to the user's cart.

#### Testing instructions

* Open this branch locally or via calypso.live (the live link is in [this comment](https://github.com/Automattic/wp-calypso/pull/54703#issuecomment-883468489))
* Navigate to `/start?flags=signup/professional-email-step`
* Pick a paid domain
* Click on the "Skip" button
* Verify that you're taken to the plan selection page/step
* Click on the "Back" link in the top left corner of the plan selection page
* Verify that you're taken to the email step
* Click on the "Skip" button again
* Select a paid plan
* Verify that you are taken to the site creation page/screen, and not sent back to the emails step
* Verify that your site is created correctly and that you're dropped into checkout with your plan and domain listed

Using the same environment as above, also work through the following flows starting from the two initial starting points:
1. Starting from `/start` (with no feature flag), verify that you can perform all of the following and see unchanged behaviour (especially not seeing the emails step!):
   - Pick the free subdomain and pick a plan - you should have your site created and the plan appear in checkout
   - Pick the free subdomain and create a free site - you should have your site created and get dropped into My Home.
   - Pick a paid domain and a paid plan - you should have your site created and both the domain and plan appear in checkout
2. Starting from `/start?flags=signup/professional-email-step`, verify that the following sequences work:
   - Pick the free subdomain and pick a plan - you should not see the emails step, your site should be created, and you should get dropped into checkout
   - Pick the free subdomain and create a free site - you should not see the emails step, your site should be created, and you should be dropped into My Home
   - Pick a paid domain, click the Skip button on the email step, and add a paid plan - you should see the emails step, the site should be created, and you should be dropped into checkout with the domain and plan listed

Related to #54313
